### DISCUSSION TITLE
When it comes to uWebSockets and uWebSockets.js, what would be the difference in performance?

### DISCUSSION THREAD

**Discussion by real2two** [uNetworking/uWebSockets#1415] [Q&A] [answered]
I'm assuming uWebSockets is faster than uWebSockets, since it's written in C++.

Just a little question of curiosity came into my mind wondering how much times better would uWebSockets be than uWebSockets.js

Since I don't really know much C++, I'm wondering if anyone here could do the actual experiment of it (determining how much times it would be faster).

**Comment by e3dio** [comment, marked as answer]
They both use the same library underneath, that is the whole point. If you call `app.publish()` it will broadcast messages the same way with the same speed. Most of any difference is all the stuff you stack on top to code your app, your JS will be less efficient depending on what you add. If you minimize your JS it can also be fast. The ease of coding with JS especially with async promises and things like worker threads with Node make it a better option for me

![image](https://user-images.githubusercontent.com/85405955/152336356-b444af63-ad34-441c-8428-ecab26a5a4c5.png)

Receiving large number of websocket message events is more efficient staying in C++ than switching between C++ and JS 
![image](https://user-images.githubusercontent.com/85405955/152727965-52b3dd65-3456-434c-b92f-c9c2da9153de.png)
